qa/common/gen_qa_ragged_models.py

```diff
@@ -471,20 +488,236 @@ def create_modelconfig(models_dir, max_batch, model_version, dtype, backend,
+        engine_bytes = engine.serialize()
+        del engine
+
+    model_name = "plan_batch_item"
+    model_version_dir = models_dir + "/" + model_name + "/" + str(model_version)
+
+    try:
+        os.makedirs(model_version_dir)
+    except OSError as ex:
+        pass  # ignore existing dir
+
+    with open(model_version_dir + "/model.plan", "wb") as f:
+        f.write(engine_bytes)
+
+    del builder
+
+
+def create_onnx_itemshape_modelfile(models_dir, model_version, dtype):
+    # Create special identity model for batch input 'BATCH_ITEM_SHAPE' testing,
+    # such model has one ragged input and one batch input, and one output to
+    # return the batch input directly. Because 'BATCH_ITEM_SHAPE' should be
+    # generated to have matching batch dimension, the output can be produced
+    # via identity op and expect Triton will scatter the output properly.
+
+    onnx_dtype = np_to_onnx_dtype(dtype)
+
+    # Create the model
+    model_name = "onnx_batch_item"
+    model_version_dir = models_dir + "/" + model_name + "/" + str(model_version)
+
+    in0_shape, idx = tu.shape_to_onnx_shape([-1], 0)
+    batch_shape, idx = tu.shape_to_onnx_shape([-1, 2], 0)
+
+    in0 = onnx.helper.make_tensor_value_info("RAGGED_INPUT", onnx_dtype,
+                                             in0_shape)
+    batch_in = onnx.helper.make_tensor_value_info("BATCH_INPUT", onnx_dtype,
+                                                  batch_shape)
+
+    batch_out_shape, idx = tu.shape_to_onnx_shape([-1, -1], idx)
+    batch_out = onnx.helper.make_tensor_value_info("BATCH_OUTPUT", onnx_dtype,
+                                                   batch_out_shape)
+
+    onnx_nodes = [
+        onnx.helper.make_node("Identity", ["BATCH_INPUT"], ["BATCH_OUTPUT"])
+    ]
+    onnx_inputs = [in0, batch_in]
+    onnx_outputs = [batch_out]
+
+    graph_proto = onnx.helper.make_graph(onnx_nodes, model_name, onnx_inputs,
+                                         onnx_outputs)
+    if FLAGS.onnx_opset > 0:
+        model_opset = onnx.helper.make_operatorsetid("", FLAGS.onnx_opset)
+        model_def = onnx.helper.make_model(graph_proto,
+                                           producer_name="triton",
+                                           opset_imports=[model_opset])
+    else:
+        model_def = onnx.helper.make_model(graph_proto, producer_name="triton")
+
+    try:
+        os.makedirs(model_version_dir)
+    except OSError as ex:
+        pass  # ignore existing dir
+
+    onnx.save(model_def, model_version_dir + "/model.onnx")
+
+
+def create_itemshape_modelconfig(models_dir, max_batch, model_version, dtype,
+                                 backend, platform):
+    version_policy_str = "{ latest { num_versions: 1 }}"
+
+    backend_spec = '''
+backend: "{}"
+'''.format(backend)
+    if backend == "tensorflow":
+        backend_spec += '''
+platform: "{}_{}"
+'''.format(backend, platform)
+
+    # Use a different model name for the non-batching variant
```

Comment:
Should this comment be somewhere else? This function seems to be for the batching variant.

Or maybe "a different model name from the non-batching variant"?